sdk-docs/how-to/flask.md

```diff
@@ -0,0 +1,103 @@
+# Add an environment variable
+
+A charm configuration can be added if you require environment variables, for
+example, to pass a token for a service. For example, add the following
+configuration in `charmcraft.yaml`:
+
+```yaml
+config:
+  options:
+    token:
+      description: The token for the service.
+      type: string
+      required: true
+```
+
+PaaS App Charmer will map this as the `FLASK_TOKEN` environment variable. In
+general, the environment variable name is `FLASK_<config option name>` where the
+config option name will be converted to upper case. The configuration can be set
+on the deployed charm using `juju config <app name> token=<token>`
+
+# Database migration
+
+If your app depends on a database it is common to run a database migration
+script before app startup which, for example, creates or modifies tables. This
+can be done by including the `migrate.sh` script in the root of your project. It
+will be executed with the same environment variables and context as the flask
+app.
+
+If the migration script fails, the app won't be started and the app charm will
+go into blocked state. The migration script will be run on every unit and it is
+assumed that it is idempotent (can be run multiple times) and that it can be run
+on multiple units at the same time without causing issues. This can be achieved
+by, for example, locking any tables during the migration.
+
```

Comment:
Is this step part of customising (a) the application, (b) the rock, or (c)the charm?

Reply:
This is related to how to run database migrations. So before the app can interact with the database, tables need to be created or changed which is typically done before the app starts. Does that make sense?